neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/Neo4jAssociationSnapshot.java

```diff
@@ -63,6 +68,28 @@ public int size() {
 	}
 
 	private static Iterable<Relationship> relationships(Node ownerNode, AssociationKey associationKey, String relationshipType) {
+		if ( relationshipType.contains( "." ) ) {
```

Comment:
Ok, I forgot to add a more expressive method. I'll update it.

Anyway this check if the collection is inside an embedded property by looking for a dot in the collection role